docs/guides/build-args.md

```diff
@@ -0,0 +1,193 @@
+        +hello | hello eggplant
+        +hello | hello world
+        output | --> exporting outputs
+```
+
+In addition to the `BUILD` command, the `--build-arg` flag can also be used with `FROM` and `COPY` commands.
+
+## Passing secrets to RUN commands
+
+Secrets are similar to build arguments; however, they are *not* defined in targets, but instead are explicitly defined for each `RUN` command that is permitted to access them.
+
+Here's an example Earthfile that accesses a secret stored under `+secrets/passwd` and exposes it under the environment variable `mypassword`:
+
+```dockerfile
+FROM alpine:latest
+hush:
+    RUN --secret mypassword=+secrets/passwd echo "my password is $mypassword"
+```
+
+{% hint style='info' %}
+It's also possible to temporarily mount a secret as a file:
+
+```dockerfile
+RUN --mount type=secret,target=/root/mypassword,id=+secrets/passwd echo "my password is $(cat /root/mypassword)"
+```
+
+The file will not be saved to the image snapshot.
+{% endhint %}
+
+The value for `+secrets/passwd` must then be supplied to earthly on the command line. Either directly via:
+
+```bash
+earthly --secret passwd=itsasecret +hush
+```
+
+or if the value is omitted, then earthly will attempt to lookup the value from an environment variable on the localhost:
+
+```bash
+passwd=itsasecret \
+earthly --secret passwd +hush
+```
+
+Running either of those commands will output:
+
+```
++hush | --> RUN echo "my password is $mypassword"
++hush | my password is itsasecret
+```
+
+{% hint style='info' %}
+### How Arguments affect caching
+
+Commands in earthly must be re-evaluated when the command itself changes (e.g. `echo "hello $name"` is changed to `echo "greetings $name"`), or when
+one of it's inputs has changed (e.g. `--build-arg name=world` is changed to `--build-arg name=banana`). Earthly creates a hash based on both the contents
+of the command and the contents of all defined arguments of the target build context.
+
+However, in the case of secrets, the contents of the secret *is not* included in the hash; therefore, if the contents of a secret changes, earthly is unable to
+detect such a change, and thus the command will not be re-evaluated.
+{% endhint %}
+
+## Storage of local secrets
+
+Earthly stores the contents of command-line-supplied secrets in memory on the localhost. When a `RUN` command that requires a secret is evaluated by BuildKit, the BuildKit
+daemon will request the secret from the earthly command-line process and will temporarily mount the secret inside the runc container that is evaluating the `RUN` command.
+Once the command finishes the secret is unmounted. It will not persist as an environment variable within the saved container snapshot. Secrets will persist in-memory
+until the earthly command exits.
+
+{% hint style='danger' %}
+Earthly and BuildKit communicate over an unencrypted TCP connection *only* when running on the same host; therefore any users on the localhost with network packet inspection
```

Comment:
IIRC it's not a TCP connection currently. We're migrating towards TCP at some point, but it'll be an encrypted connection. @dchw has more details.

Reply:
that's a good point, the default is for socket-based connection: `default: "docker-container://earthly-buildkitd"`; I'll remove this.